rpc/src/main_rpc_server.cpp

```diff
@@ -21,106 +21,89 @@
-using namespace uprotocol::uuid;
-using namespace uprotocol::uri;
-using namespace uprotocol::client;
+using namespace uprotocol::v1;
+using namespace uprotocol::communication;
+using namespace uprotocol::datamodel::builder;
 
 bool gTerminate = false;
 
 void signalHandler(int signal) {
-    if (signal == SIGINT) {
-        std::cout << "Ctrl+C received. Exiting..." << std::endl;
-        gTerminate = true; 
-    }
+	if (signal == SIGINT) {
+		std::cout << "Ctrl+C received. Exiting..." << std::endl;
+		gTerminate = true;
+	}
 }
 
-class RpcListener : public UListener {
-
-    public:
-       
-         UStatus onReceive(UMessage &message) const override {
-            /* Construct response payload with the current time */
-            auto currentTime = std::chrono::system_clock::now();
-            uint64_t currentTimeMilli =
-                std::chrono::duration_cast<std::chrono::milliseconds>(
-                        currentTime.time_since_epoch()).count();
-
-            UPayload responsePayload(
-                    reinterpret_cast<const uint8_t*>(&currentTimeMilli),
-                    sizeof(currentTimeMilli),
-                    UPayloadType::VALUE);
-
-            /* Build response attributes - the same UUID should be used to send the response 
-             * it is also possible to send the response outside of the callback context */
-            auto builder = UAttributesBuilder::response(
-                    message.attributes().sink(),
-                    message.attributes().source(),
-                    message.attributes().priority(),
-                    message.attributes().id());
-
-            UAttributes responseAttributes = builder.build();
-            UMessage messageResp(responsePayload, responseAttributes);
-
-            /* Send the response */
-            auto ret =  UpZenohClient::instance()->send(messageResp);
-            return ret;
-        }
-};
-
-/* The sample RPC server applications demonstrates how to receive RPC requests and send a response back to the client -
- * The response in this example will be the current time */
-int main(int argc, 
-         char** argv) {
-
-    (void)argc;
-    (void)argv;
-    
-    RpcListener listener;
-
-    signal(SIGINT, signalHandler);
-
-    UStatus status;
-    std::shared_ptr<UpZenohClient> transport = UpZenohClient::instance(
-            BuildUAuthority().setName("device1").build(),
-            BuildUEntity().setName("rpc.client").setMajorVersion(1).setId(1).build());
-
-    /* init zenoh utransport */
-    if (nullptr == transport) {
-        spdlog::error("UpZenohClient init failed");
-        return -1;
-    }
+std::optional<uprotocol::datamodel::builder::Payload> OnReceive(
+    const UMessage& message) {
+	// Validate message is an RPC request
+	if (message.attributes().type() != UMessageType::UMESSAGE_TYPE_REQUEST) {
+		spdlog::error("Received message is not a request\n{}",
+		              message.DebugString());
+		return {};
+	}
```

Comment:
I _think_ RPC server already does this check

Reply:
I wasn't sure if it did or not, but seeing as this is an example program, I didn't see the harm in putting in the check.  Maybe I should change the comment to something like:

// Validate message is an RPC request - should not be necessary